lib/jsonapi/plugs/query_parser.ex

```diff
@@ -246,7 +246,7 @@ defmodule JSONAPI.QueryParser do
 
   @spec get_view_for_type(module(), String.t()) :: module() | no_return()
   def get_view_for_type(view, type) do
-    case Enum.find(view.relationships, fn {k, _v} -> Atom.to_string(k) == type end) do
+    case Enum.find(view.relationships, fn {_k, v} -> v.type == type end) do
```

Comment:
It looks to me that although we're matching the spec more closely with this change that it _could_ break existing user's apps. I think we'll need to support looking for a view on both the `type` and key for now, and log a deprecation warning that we'll only support the `type` parameter in 2.0.

We don't have a formal deprecation logging mechanism in play right now, but in the past we've done this:

https://github.com/jeregrine/jsonapi/commit/323eab356ff309e7d383c63d7f612d662771fd72#diff-145509c96e7086ca2b927dd4897ed875L10

Reply:
Added deprecation warning and restored original behavior (to run second after the preferred behavior)